src/elm/Page/Shop.elm

```diff
@@ -254,25 +255,22 @@ viewShopFilter loggedIn filter =
 viewGrid : LoggedIn.Model -> List Card -> Model -> Html Msg
 viewGrid loggedIn cards model =
     let
-        v_ viewFn card =
-            viewFn model loggedIn card
+        v_ viewFn index card =
+            viewFn model loggedIn card index
```

Comment:
This `v_` was quite confusing for me... Maybe we could make it simpler?

If we switch `Int` and `Card` arguments in `viewCard` function then currying will do the work:

```elm
viewGrid : LoggedIn.Model -> List Card -> Model -> Html Msg
viewGrid loggedIn cards model =
    div [ class "flex flex-wrap -mx-2" ]
        (List.indexedMap
            (viewCard model loggedIn)
            cards
        )


viewCard : Model -> LoggedIn.Model -> Int -> Card -> Html Msg
viewCard model ({ shared } as loggedIn) index card =
    -- ...
```

Reply:
Good catch, I didn't know we could do this :)